da/test/da_test.go

```diff
@@ -205,35 +231,53 @@ func doTestRetrieve(t *testing.T, dalc da.DataAvailabilityLayerClient) {
 	time.Sleep(mockDaBlockTime + 20*time.Millisecond)
 
 	retriever := dalc.(da.BlockRetriever)
-	countAtHeight := make(map[uint64]int)
-	blocks := make(map[*types.Block]uint64)
+	countHeadersAtHeight := make(map[uint64]int)
+	countDataAtHeight := make(map[uint64]int)
+	blocks := make(map[*types.Block]DAHeights)
 
 	for i := uint64(0); i < 100; i++ {
 		b := getRandomBlock(i, rand.Int()%20) //nolint:gosec
-		resp := dalc.SubmitBlock(ctx, b)
+		heights := DAHeights{}
+		resp := dalc.SubmitBlockHeader(ctx, &b.SignedHeader)
 		assert.Equal(da.StatusSuccess, resp.Code, resp.Message)
+		countHeadersAtHeight[resp.DAHeight]++
+		heights.header = resp.DAHeight
```

Comment:
this will work when header and block have same height as well. the test was giving me different heights hence i had to separately handle them. however, if they have the same height, it should work just fine.